sound/soc/sof/topology.c

```diff
@@ -1041,6 +1041,13 @@ static int sof_control_load_volume(struct snd_soc_component *scomp,
 		goto out;
 	}
 
+	/*
+	 * if control has more than 2 channels, we need to override the info as
+	 * asoc layer dapm controls support only mono and stereo.
```

Comment:
no its more because the pre-defined mixer control types only support upto stereo channels. As far as the control definition goes, it can have up to 8 channels

Reply:
ok. i'll refine this. I think this is still correct, in a way. I know that kcontrol supports more channels, but dapm restricts it to max 2. So I mean you can define control to have bigger channel count, but most control related functions in soc/dapm layer just don't want to co-operate if you have > 2 channels.

Reply:
tried to make this better, but probably still confusing to people not familiar with the issue...